source/_patterns/01-elements/input/input.scss

```diff
@@ -1,50 +1,159 @@
 @charset "utf-8";
 
 @use "@db-ui/base/build/scss/variables" as *;
+@use "@db-ui/base/build/scss/variables.global" as *;
 @use "@db-ui/base/build/scss/helpers/functions" as *;
-@import "input.variables";
-@import "../form-elements";
+@use "@db-ui/base/build/scss/icon/icons.helpers" as *;
+
+$icons-path: "../../../icons/" !default;
+
+@mixin label-focus-animation(
+	$translationX: var(--elm-input-padding-inline-start)
+) {
+	$transformScale: -1.25;
+
+	transition: opacity 0.08s, transform 0.15s;
+	opacity: var(--db-input-focus-opacity);
+
+	transform: translate($translationX, calc($db-sizing-md * $transformScale))
+		scale(var(--db-input-label-scale));
+}
 
 .elm-input {
-	@extend %form-element;
+	--elm-input-padding-inline-start: #{$db-spacing-fixed-sm};
+	--elm-input-padding-inline-end: #{$db-spacing-fixed-sm};
 
-	color: $input---color;
+	position: relative;
 
-	font-size: to-rem($pxValue: 16);
-	height: to-rem($pxValue: 44);
-	line-height: to-rem($pxValue: 24);
-	max-width: 100%;
+	// padding-inline-end according to current amount of icons
+	&[data-icon-before] {
+		--elm-input-padding-inline-start: #{calc(
+				2 * $db-spacing-fixed-sm + 0.425rem
+			)};
+
+		input {
+			padding-inline-start: var(--elm-input-padding-inline-start);
+		}
+	}
+
+	&[data-icon-after],
+	&[data-variant] {
+		--elm-input-padding-inline-end: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem
+			)};
+
+		input {
+			padding-inline-end: var(--elm-input-padding-inline-end);
+		}
+	}
 
-	padding: $db-spacing-fixed-regular-xs $db-spacing-fixed-regular-m;
+	&[data-variant][data-icon-after] {
+		--elm-input-padding-inline-end: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem + 0.75rem + #{$db-spacing-fixed-xs}
+			)};
 
+		input {
+			padding-inline-end: var(--elm-input-padding-inline-end);
+		}
+
+		.elm-state-icon::after {
+			margin-right: calc(0.75rem + $db-spacing-fixed-xs);
```

Comment:
You could use `margin-inline-end` as an equivalent to `padding-inline-end` here as well.

Reply:
@mfranzke done